openml/extensions/sklearn/extension.py

```diff
@@ -1641,6 +1641,22 @@ def _prediction_to_probabilities(
             if X_test is None:
                 raise TypeError("argument X_test must not be of type None")
 
+        try:
+            # check if model is fitted
+            # 'predict' internally calls sklearn.utils.validation.check_is_fitted for every
+            # model-specific attribute it excepts, thus offering a more robust check than
+            # a generic simplified call of check_is_fitted(model_copy)
+            from sklearn.exceptions import NotFittedError
+
+            model.predict(X_train)
+            warnings.warn(
+                "The model is already fitted!"
+                " This might cause inconsistency in comparison of results."
+            )
+        except NotFittedError:
+            # model is not fitted, as is required
+            pass
```

Comment:
I thought in the Python call we discussed that perhaps we would check this at the first call to ``run_model_on_task``?
In either case I would extract this to a separate method `_raise_warning_if_fitted` to make sure the functions don't get too big (they already are).

Reply:
Alright, will make it into a function and push.

As for its placement, I reconsidered it given that irrespective of what is called, `run_model_on_task` or `run_flow_on_task`, this function is what the call is reduced to. Hence went ahead with this placement for this snippet of code.

Reply:
`run_model_on_task` actually calls `run_flow_on_task`. That said, then we would need to add a function to the extension interface that will indicate if a model is already fit, otherwise we can't check it in a general way.

Reply:
@mfeurer do you think this is something we should want? or do we just leave it to the extension devs to implement a warning if they see it fit?

Reply:
I'd leave it to the extensions devs. For scikit-learn I'd use [this function](https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_is_fitted.html).